base/REPLCompletions.jl

```diff
@@ -161,7 +161,13 @@ function complete_path(path::AbstractString, pos; use_envpath=false)
                 continue
             end
 
-            local filesinpath = readdir(pathdir)
+            local filesinpath
+            try
+                filesinpath = readdir(pathdir)
+            catch
+                # Bash allows dirs in PATH that can't be read, so we should as well.
```

Comment:
This really should not be a `try` `catch`, but should check if the pathdir is readable and only then calling `readdir()`